gazebo_grasp_plugin/include/gazebo_grasp_plugin/GazeboGraspFix.h

```diff
@@ -100,12 +100,12 @@ class GazeboGraspFix : public ModelPlugin
      * Gets called just after the object has been attached to the palm link on \e armName
      */
     virtual void OnAttach(const std::string &objectName,
-                          const std::string &armName) {}
+                          const std::string &armName);
```

Comment:
I see you put the publication of the event here (in the cpp file of this). In principle this is fine, because this function wasn't currently doing anything and I'm not aware of anyone using subclasses of this. However if we're going to use this, we must at least remove the `virtual`.
Alternatively, we could add a protected non-virtual `DoOnAttach()` which calls `OnAttach`, and we do the publishing in the `DoOnAttach`, which also calls `OnAttach`. This way we don't change the logic of this interface.
Current calls to `OnAttach` in the cpp file would then need to be replaced by call to `DoOnAttach`.

Same applies to `OnDetach`.

I'd tend to favor the `DoOnAttach` solution but I'm open for discussion.

Reply:
Ah shoot I forgot to adjust this in my PR to your branch... I will update it later with this

Reply:
I just fixed up my PR to your branch, made this private and non-virtual. If somebody is using it in subclasses (which I doubt) this will have to be changed later again.